### PR TITLE
Xsuaa activation - steps

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- implementation of a generic offering step and use it for xsuaa
- implementation of XSUAA steps: provisioning, deprovisioning, bind and unbind

What is missing:
- setting proper overrides
- storing binding data (in the operation, as an encrypted value)

**Related issue(s)**
https://github.tools.sap/kyma/backlog/issues/908

### CODE REVIEW

File: components/kyma-environment-broker/internal/process/provisioning/service_manager_offering_step.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	if err != nil {
+		return s.handleError(operation, err, "unable to get Service Manager offerings", log)
+	}
+	if len(offerings.ServiceOfferings) != 1 {
+		return s.operationManager.OperationFailed(operation,
+			fmt.Sprintf("expected one %s Service Manager offering, but found %d", s.offeringName, len(offerings.ServiceOfferings)))
+	}
+	info.ServiceID = offerings.ServiceOfferings[0].CatalogID
+	info.BrokerID = offerings.ServiceOfferings[0].BrokerID
+	log.Infof("Found offering: catalogID=%s brokerID=%", info.ServiceID, info.BrokerID)
```
Comment (ksputo) [inline components/kyma-environment-broker/internal/process/provisioning/service_manager_offering_step.go:66]:
```suggestion
	log.Infof("Found offering: catalogID=%s brokerID=%s", info.ServiceID, info.BrokerID)
```
fix string formatting

File: components/kyma-environment-broker/internal/process/provisioning/xsuaa_provisioning_step.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+func (s *XSUAAProvisioningStep) Run(operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	if operation.XSUAA.Instance.ProvisioningTriggered {
+		return operation, 0, nil
+	}
+
+	smCli, err := operation.ServiceManagerClient(log)
+	if err != nil {
+		return s.handleError(operation, err, "unable to create Service Manage client", log)
```
Comment (ksputo) [inline components/kyma-environment-broker/internal/process/provisioning/xsuaa_provisioning_step.go:39]:
```suggestion
		return s.handleError(operation, err, "unable to create Service Manager client", log)
```

File: components/kyma-environment-broker/internal/model.go [hunk 3]
```diff
@@ -418,3 +445,16 @@ func (l ComponentConfigurationInputList) DeepCopy() []*gqlschema.ComponentConfig
 	}
 	return copiedList
 }
+
+func serviceManagerRequestCereds(parameters ProvisioningParameters) *servicemanager.Credentials {
```
Comment (ksputo) [inline components/kyma-environment-broker/internal/model.go:449]: I suppose it should be
```suggestion
func serviceManagerRequestCreds(parameters ProvisioningParameters) *servicemanager.Credentials {
```

File: components/kyma-environment-broker/internal/process/deprovisioning/xsuaa_deprovision_step.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+func NewXSUAADeprovisionStep(os storage.Operations) *XSUAADeprovisionStep {
+	return &XSUAADeprovisionStep{
+		operationManager: process.NewDeprovisionOperationManager(os),
+	}
+}
+
+var _ Step = (*XSUAADeprovisionStep)(nil)
+
+func (s *XSUAADeprovisionStep) Name() string {
+	return "XSUAA_Unbind"
```
Comment (radufa) [inline components/kyma-environment-broker/internal/process/deprovisioning/xsuaa_deprovision_step.go:27]: Please change the name

**Review by ksputo — APPROVED** [review]

**Pull request merged**